### ISSUE TITLE
robot reason using ELK doesn't seem to spot inconsistent classes

### ISSUE THREAD

**Issue by marieALaporte** [ontodev/robot#535]
I am using the 0.5.0 SNAPSHOT version of the ELK reasoner on Protégé. When running ELK on Protégé on the envo release file, I got some inconsistent classes that robot did not pick up during the release process. @pbuttigieg fixed the issues on the envo-edit.owl file in this [commit](https://github.com/EnvironmentOntology/envo/pull/815/commits/414e52c5ef443f7d5e9bc28c9759b9fef77d87d4).

I just built robot from source and still get no error when I reason over the envo.owl file (release file).

Any idea why ELK behave differently? Thanks!

**Comment by balhoff** [comment]
ELK 0.5.0 supports (some) reasoning with object property ranges, which the main release (0.4.3) does not. So this could be uncovering hidden errors, e.g.:

https://github.com/geneontology/go-ontology/issues/16524

https://github.com/obophenotype/uberon/issues/1455

https://github.com/monarch-initiative/mondo/issues/439

**Comment by marieALaporte** [comment]
Thanks @balhoff. So the recommendation is to use the reasoner in Protégé before releasing an ontology? and of course to use the ELK 0.5.0 version in Protégé?
Is there any plan for robot (or owl api maybe) to embed ELK 0.5.0?

**Comment by jamesaoverton** [comment]
Yes, ROBOT uses ELK 0.4.3, and we've been waiting for some of these problems to be sorted out before upgrading to 0.5.0. You could build your own ROBOT with ELK 0.5.0 by changing the dependency here: https://github.com/ontodev/robot/blob/master/pom.xml#L173. You might also have to specify which axiom types to generate with `reason`, see http://robot.obolibrary.org/reason#generated-axioms.
